runtime/src/deterministic.rs

```diff
@@ -102,15 +102,18 @@ impl Metrics {
     }
 }
 
+/// A SHA-256 digest.
+type Digest = [u8; 32];
+
 /// Track the state of the runtime for determinism auditing.
 pub struct Auditor {
-    hash: Mutex<Vec<u8>>,
+    digest: Mutex<Digest>,
```

Comment:
This change was unrelated but couldn't resist removing heap allocs from this hot path 👀